packages/editor/src/components/rich-text/format-toolbar/link-container.js

```diff
@@ -19,95 +23,205 @@ import { filterURLForDisplay } from '../../../utils/url';
 
 const stopKeyPropagation = ( event ) => event.stopPropagation();
 
-const LinkContainer = ( props ) => {
-	const {
-		editLink,
-		formats,
-		isEditing,
-		isPreviewing,
-		linkValue,
-		onChangeLinkValue,
-		onKeyDown,
-		opensInNewWindow,
-		selectedNodeId,
-		setLinkTarget,
-		settingsVisible,
-		submitLink,
-		toggleLinkSettingsVisibility,
-	} = props;
-
-	const linkSettings = settingsVisible && (
-		<div className="editor-format-toolbar__link-modal-line editor-format-toolbar__link-settings">
-			<ToggleControl
-				label={ __( 'Open in New Window' ) }
-				checked={ opensInNewWindow }
-				onChange={ setLinkTarget } />
-		</div>
-	);
-
-	return (
-		<Fill name="RichText.Siblings">
-			<PositionedAtSelection className="editor-format-toolbar__link-container">
-				<Popover
-					position="bottom center"
-					focusOnMount={ isEditing ? 'firstElement' : false }
-					key={ selectedNodeId /* Used to force rerender on change */ }
+class LinkContainer extends Component {
+	constructor( { link, selection } ) {
+		super( ...arguments );
+
+		this.editLink = this.editLink.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.onChangeInputValue = this.onChangeInputValue.bind( this );
+		this.toggleLinkSettingsVisibility = this.toggleLinkSettingsVisibility.bind( this );
+		this.setLinkTarget = this.setLinkTarget.bind( this );
+
+		this.state = {
+			inputValue: link && link.attributes && link.attributes.href ? null : '',
+			opensInNewWindow: null,
+			lastSelection: selection,
+			selectionKey: 0,
+		};
+	}
+
+	static getDerivedStateFromProps( { link, selection }, state ) {
+		if ( selection === state.lastSelection ) {
+			return null;
+		}
+
+		return {
+			inputValue: link.attributes && link.attributes.href ? null : '',
+			opensInNewWindow: null,
+			lastSelection: selection,
+			selectionKey: state.selectionKey + 1,
+		};
+	}
+
+	onKeyDown( event ) {
+		if ( event.keyCode === ESCAPE ) {
+			event.stopPropagation();
+			this.props.removeFormat( 'a' );
```

Comment:
When you call `removeFormat` it will update props so it will use `getDerivedStateFromProps` to update state, too. In theory, you shouldn't need to reset state explicitly. I saw a similar logic in other places in this file. It was updated in `master`. 

@noisysocks can you double check that your changes for the url still work properly?